### PR TITLE
mis_builder_analytic: New module

### PR DESCRIPTION
**Pull request by ernestotejeda**
Depends on:

- [x] https://github.com/OCA/mis-builder/pull/149

Cc @Tecnativa
TT14942

### CODE REVIEW

File: mis_builder_analytic/README.rst [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+**Table of contents**
+
+.. contents::
+   :local:
+
+Usage
+=====
+
+To use this module, you need to:
```
Comment (elicoidal) [inline mis_builder_analytic/README.rst:39]: Please add a screenshot

File: mis_builder_analytic/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+{
+    'name': 'MIS Builder Analytic',
+    'summary': "Provide account analytic lines for MIS builder reports",
+    'version': '11.0.1.0.0',
+    'license': 'AGPL-3',
+    'author': 'Tecnativa, '
+              'Odoo Community Association (OCA)',
+    'website': 'https://github.com/OCA/mis-builder',
+    'depends': [
```
Comment (elicoidal) [inline mis_builder_analytic/__manifest__.py:12]: for the sake of the apps-store, please add an image or logo

**Review by elicoidal — APPROVED** [review]
LGTM. some details to improve the apps store output

**Comment by elicoidal** [comment]
Maybe you can add an entry in https://github.com/OCA/mis-builder/tree/12.0/docs too as this is a very nice feature

**Comment by sbidoul** [comment]
Hi, thanks for this contribution.

I'm too sure of the kind of reports this enables, that would not be possible with regular move lines.
Indeed, this view joins on account_move_line to find the account, and if the analytic line has no related move line (eg for timesheets), there will be no account and the report will not work.

Could you elaborate your use case?

File: mis_builder_analytic/models/mis_account_analytic_line.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+                      ELSE 0.0
+                    END AS credit,
+                    CASE
+                      WHEN aal.amount < 0 THEN (aal.amount * -1)
+                      ELSE 0.0
+                    END AS debit,
+                    aal.amount as balance
+                FROM
+                    account_analytic_line aal
+                    INNER JOIN account_move_line aml ON aml.id = aal.move_id
```
Comment (pedrobaeza) [inline mis_builder_analytic/models/mis_account_analytic_line.py]: It must be `LEFT JOIN`

**Comment by pedrobaeza** [comment]
@sbidoul I have put it the comment that it should be `LEFT JOIN`

**Comment by sbidoul** [comment]
Ah, no sorry, part of my comment above is wrong. With INNER JOIN, if the analytic line has no move_id, it will be left out of the view, so account_id will always have a value in the view, that's fine.

Still, I've the feeling a mis report on this view will give the same result as a report on move lines, no?

**Comment by pedrobaeza** [comment]
No, the idea is to split by analytic accounts, not general accounts. I haven't reviewed yet this work, but probably the problem is that the main general account mustn't be used, but the analytic account.

File: mis_builder_analytic/models/mis_account_analytic_line.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    _name = "mis.account.analytic.line"
+    _auto = False
+    _description = "MIS Account Analytic Line"
+
+    date = fields.Date()
+    analytic_line_id = fields.Many2one(
+        string="Analytic entry",
+        comodel_name='account.analytic.line',
+    )
+    account_id = fields.Many2one(
```
Comment (pedrobaeza) [inline mis_builder_analytic/models/mis_account_analytic_line.py:17]: Indeed, this is the problem... We don't want the main account. What we want is to have the analytic account.

**Comment by pedrobaeza** [comment]
@ernestotejeda please make here the changes needed and include in oca_dependencies.txt the branch of #149 for proper testing

**Comment by ernestotejeda** [comment]
@sbidoul i have done the changes proposed

**Comment by ernestotejeda** [comment]
@pedrobaeza I have updated this PR with the changes made in https://github.com/OCA/mis-builder/pull/149

**Comment by pedrobaeza** [comment]
This one includes the commit from the other PR for being testable.

**Comment by pedrobaeza** [comment]
hehe, we have met in the same clarifier comment.

**Comment by rafaelbn** [comment]
### Objetive:

The possibility to create MIS REPORTS based in `account.analytic.line` filtering and groping by the fields:

- `account_id` | Analytic Account
- `tag_ids` | Tags
- `general_account_id` | Financial Account
- .... and in the future all the fields of the model `account.analytic.line` like Department, Product, etc...

### Why?

Because the cost of the employee's time-sheets are NOT in account move lines

We will see how to solve this misunderstanding

### Example:

- We sale and invoice (so we have account move lines and account analytic lines) in "project 1" ( this is a project and the analytic account) "product A" and "product B". So we have

Product | Analtytic Account | Analytic tag | Amount
------------ | ------------- | ------------- | -------------
Product A | Project 1 | Tag Segment 1 | 1000
Product B | Project 1 | Tag Segment 2 | 2000

- We make **timesheet** (only account analytic lines) in "project 1" of 2 employees. Employee A cost is 50 per hour and Employee B is 80 per hour

Product | Analtytic Account | **Hours** | Analytic tag | Amount
------------ | ------------- | ------------- | ------------- | -------------
Employee A | Project 1 | 2 | Tag Segment 1 | 100
Employee B | Project 1 | 3 | Tag Segment 2 | 240

- So we expect a MIS Builder Inform that based in account.analytic.line says

1. Tag Segment + Analytic account

Tag | Analtytic Account | Type | Amount
------------ | ------------- | ------------- | -------------
Tag segment 1 | Project 1 | Profit | 900
------------ | ------------- | ------------- | -------------
_ | Project 1 | Income | 1000
_ | Project 1 | Cost | 100
------------ | ------------- | ------------- | -------------
Tag segment 2 | Project 1 | Profit | 1.760
------------ | ------------- | ------------- | -------------
_ | Project 1 | Income | 2000
_ | Project 1 | Cost | 240


2. Analytic account + Tag Segment

Analtytic Account | Tag | Type | Amount
------------ | ------------- | ------------- | -------------
Project 1 | _ | Profit | 2.660
------------ | ------------- | ------------- | -------------
_ | Tag segment 1 | Profit | 900
_ | Tag segment 2 | Profit | 1.760


3. Expected and desired is algo be possible to filter and group by - `general_account_id` | Financial Account

:cry: 

high-res

**Comment by pedrobaeza** [comment]
@rafaelbn please don't come with offtopics, as this is not the aim of this PR, and can't be without major refactoring on mis_builder.

**Review by rafaelbn — APPROVED** [review]

**Comment by rafaelbn** [comment]
Please @erick-tejada This branch has conflicts that must be resolved thanks

**Comment by rafaelbn** [comment]
Please @ernestotejeda This branch has conflicts that must be resolved

File: mis_builder_analytic/models/mis_account_analytic_line.py [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+                    aal.company_id as company_id,
+                    'posted'::VARCHAR as state,
+                    CASE
+                      WHEN aal.amount >= 0.0 THEN aal.amount
+                      ELSE 0.0
+                    END AS credit,
+                    CASE
+                      WHEN aal.amount < 0 THEN (aal.amount * -1)
+                      ELSE 0.0
+                    END AS debit,
```
Comment (sbidoul) [inline mis_builder_analytic/models/mis_account_analytic_line.py:58]: A limitation of these views is that debit and credit must be in company currency.

**Comment by sbidoul** [comment]
@ernestotejeda May I ask you to move this PR to OCA/account-financial-reporting or OCA/mis-builder-contrib? Indeed I don't plan to maintain it as part of the core mis-builder repo where I try to keep feature parity across several odoo versions. Thanks!

**Comment by pedrobaeza** [comment]
@ernestotejeda move it to https://github.com/OCA/mis-builder-contrib

**Pull request closed without merging**